src/libp2p_config.erl

```diff
@@ -236,7 +236,7 @@ insert_listener(TID, Addrs, Pid) ->
 lookup_listener(TID, Addr) ->
     lookup_pid(TID, ?LISTENER, Addr).
 
--spec remove_listener(ets:tab(), string()) -> true.
+-spec remove_listener(ets:tab(), string() | undefined) -> true.
```

Comment:
ideally we remove the calls when the listener addr is undefined instead

Reply:
Since the state record permits address to be undefined, that could be anywhere it is passed unchecked, like https://github.com/helium/erlang-libp2p/blob/master/src/relay/libp2p_relay_server.erl#L199